x-pack/plugins/apm/server/routes/service_groups/route.ts

```diff
@@ -188,12 +154,57 @@ const serviceGroupServicesRoute = createApmServerRoute({
     return { items };
   },
 });
+type ServiceGroupCounts = Record<string, { services: number; alerts: number }>;
+const serviceGroupCountsRoute = createApmServerRoute({
+  endpoint: 'GET /internal/apm/service-group/counts',
+  options: {
+    tags: ['access:apm'],
+  },
+  handler: async (resources): Promise<ServiceGroupCounts> => {
+    const { context } = resources;
+    const {
+      savedObjects: { client: savedObjectsClient },
+    } = await context.core;
+
+    const [serviceGroups, authorizedAlertsIndices, apmEventClient] =
+      await Promise.all([
+        getServiceGroups({ savedObjectsClient }),
+        getAlertsIndices(resources),
+        getApmEventClient(resources),
+      ]);
+
+    const [servicesCounts, serviceGroupAlertsCount] = await Promise.all([
+      getServicesCounts({
+        apmEventClient,
+        serviceGroups,
+        start: datemath.parse('now-24h')!.toDate().getTime(),
+        end: datemath.parse('now')!.toDate().getTime(),
+      }),
+      getServiceGroupAlerts({
+        serviceGroups,
+        authorizedAlertsIndices,
+        context,
+      }),
+    ]);
+    const serviceGroupCounts: ServiceGroupCounts = serviceGroups.reduce(
+      (acc, { id }): ServiceGroupCounts => ({
```

Comment:
I think you can instead type the reduce.

```suggestion
    const serviceGroupCounts: ServiceGroupCounts = serviceGroups.reduce<ServiceGroupCounts>(
      (acc, { id })  => ({
```

Reply:
changing this results in a linting error:
```
190:7  error  Missing return type on function  @typescript-eslint/explicit-function-return-type
```